apps/pp/utils.py

```diff
@@ -89,3 +89,60 @@ def set_relation(self, resource_name, resource_id, relation_name=None):
                 'data': data,
                 'links': self.root_obj.id
             }
+
+
+OMITTED_QUERY_VARS = (
+    'utm_campaign',
+    'utm_medium',
+    'utm_term',
+    'utm_name',
+    'utm_source',
+)
+
```

Comment:
I'm not sure this is the best approach and it is too optimistic to assume we can discard parameters based on their names alone (and that it is a significant improvement over the "discard-no-parameters" policy).

I think it's the right time to improve on what you've already developed, but since I already started wondering, I'm leaving for the future 3 other solutions that came to my mind.
- **domain + parameter**: discard parameters based on the domain + parameter name
- **discard by default**: discard all parameters except those that typically modify the page (such as `page=1`). Rely on the frontend side to ignore the annotations mismatched with the site content.
- [forget for now :) ] **actual comparison**: simple HTML comparison to determine url "sameness"

Reply:
- discard by default: this one really scares me because we either build good database or break many sites
- domain + parameter: this would require much better database, not to mention implementing it

For now I would stay simple as long as it works, so in this situation it is accepting all parameters. With one minor change: only blacklisting most obvious, very popular ones. I started with `utm_` prefix as it  is very strong convention (https://en.wikipedia.org/wiki/UTM_parameters) for variables that do not affect content and are consumed by different analytics tools. They are used quite frequently (I guess use click link with utm every day), so we should and can 100 % safely exclude them.

Probably some more of that kind is `ref`, used for the same purpose referrals. I argue for blacklisting such vars, because there are relatively few while people really do come across them very often in this days when $ on the Internet depends on accurate tracking source of site conversions.